### PR TITLE
Reorganize test folder

### LINKED ISSUE #109: Reorganize tests folder and adjust `make` commands
**Underlying concern**
Windows currently bypasses parts of the tests because the property testing with hypothesis exceeds the time limits on the builds. This is particular just to how windows handles the virtual environments and is not indicative of other parts of the unit tests which pass. The property tests need to be isolated so they can still be run on Linux and MacOS.

- [ ] **Update Make Commands**
* Only property tests : `make property-tests`
* Only integrations tests: `make integration-tests`
* Only unit tests: `make unit-tests`
* All tests: `make test`
* Coverage for all tests: `make coverage`

- [ ] **Update Folder**
📂  tests
---- 📂  integration
---- 📂  property
---- 📂  unit

- [ ] **Update Continuous Integration**

### PR DESCRIPTION
**Pull request by TuTomasz**
### Issue
Fixes #109 

### Description
- Restructured the test directory-based to include property, unit and integration tests. 
- Updated added makefile commands to run each test group individually using commands (unit-tests, property-tests,integration-tests) and modified test command to run the above in series.

**Known issue:** Running test_decryption_mediator.py separately but part of the property group as during a coverage run causes failures. It is also the most long-running of the tests. This arrangement avoids the problem. 

**Aditional:** Did not update workflow files in this PR wanted to make sure I have the test arrangement correct.

### Testing
Code Review

### Checklist
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] 🤔 **CONSIDER** adding a unit test if your PR resolves an issue.
- [x] ✅ **DO** check open PR's to avoid duplicates.
- [x] ✅ **DO** keep pull requests small so they can be easily reviewed.
- [x] ✅ **DO** build locally before pushing.
- [x] ✅ **DO** make sure tests pass.
- [x] ✅ **DO** make sure any new changes are documented.
- [x] ✅ **DO** make sure not to introduce any compiler warnings.
- [x] ❌**AVOID** breaking the continuous integration build.
- [x] ❌**AVOID** making significant changes to the overall architecture.

💚Thank you!

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -85,9 +85,24 @@ validate:
 	@pipenv run python -c 'import electionguard; print(electionguard.__package__ + " successfully imported")'
 
 # Test
+unit-tests:
+	@echo ✅ UNIT TESTS
+	pipenv run pytest tests/unit
+
+property-tests:
+	@echo ✅ PROPERTY TESTS
+	pipenv run pytest tests/test_decryption_mediator.py
```
Comment (keithrfung) [inline Makefile:94]: Is this one isolated due to the tests? Also linting is failing but it actually just looks like non used imports?

**Review by keithrfung — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (TuTomasz) [reply Makefile:94]: If placed with the other property tests and run via "make test" everything passes. However, causes failures when "make coverage" runs. This issue does not happen when the test runs in isolation or ahead of other property tests. That is why I modified the property-test group this way.

**Comment by keithrfung** [comment]
In my #216 PR I'm gonna be adding tests, this may require you to move them after this. Just FYI.

**Comment by TuTomasz** [comment]
Would you like any of these test categories to be integrated into the pull_request.yml or release.yml under mac and windows?

**Comment by keithrfung** [comment]
@TuTomasz Missed that question. The answer is the Mac and Windows should ONLY be running the integration tests.

**Comment by TuTomasz** [comment]
@keithrfung Given the way the tests are separated now It might be beneficial to run the unit test category to on Mac and Windows. That test group runs super fast and I don't think it will give you Issues on Github actions. Currently, the last commit addresses only the requirements in the original issue post. I didn't modify the yml files.

**Comment by keithrfung** [comment]
@TuTomasz Do you mind taking a look at conflicts so I can pull this in? Sorry for delay.

**Comment by TuTomasz** [comment]
> @TuTomasz Do you mind taking a look at conflicts so I can pull this in? Sorry for delay.

**Review by keithrfung — APPROVED** [review]

**Pull request merged**